File: NasleGhalam.ViewModels/Exam/ExamViewModel.cs
using NasleGhalam.ViewModels._Attributes;
using System;
using System.ComponentModel.DataAnnotations;

namespace NasleGhalam.ViewModels.Exam
{
    public class ExamViewModel
    {

        public int Id { get; set; }


        [Display(Name = "نام")]
        [Required(ErrorMessageResourceType = typeof(ErrorResources), ErrorMessageResourceName = "Required")]
        [MaxLength(50, ErrorMessageResourceType = typeof(ErrorResources), ErrorMessageResourceName = "MaxLen")]
        public string Name { get; set; }

        //TODO: complate data anotation
        [Display(Name = "تاریخ امتحان")]
        [Required(ErrorMessageResourceType = typeof(ErrorResources), ErrorMessageResourceName = "Required")]
        public DateTime Date { get; set; }

        [Display(Name = "گروه درسی")]
        public string EducationGroupName { get; set; }

        [Display(Name = "گروه درسی")]
        [RequiredDdlValidator(invalidValue: "0", ErrorMessageResourceType = typeof(NasleGhalam.ViewModels.ErrorResources), ErrorMessageResourceName = "RequiredDll")]
        public int EducationGroupId { get; set; }

        [Display(Name = "سال تحصیلی")]
        public string EducationYearName { get; set; }

        [Display(Name = "سال تحصیلی")]
        [RequiredDdlValidator(invalidValue: "0", ErrorMessageResourceType = typeof(NasleGhalam.ViewModels.ErrorResources), ErrorMessageResourceName = "RequiredDll")]
        public int EducationYearId { get; set; }


    }
}

File: NasleGhalam.ClientApp/src/viewModels/examViewModel.js
import {
  required,
  displayName,
  maxLength,
  numeric,
  requiredDdl
} from 'plugins/vuelidate';

export default {
  examObj: {
    Name: {
      displayName: displayName('نام'),
      maxLength: maxLength(50),
      required
    },
    Date: {
      displayName: displayName('تاریخ امتحان'),
      maxLength: maxLength(50),
      required
    },
    EducationGroupName: {
      displayName: displayName('گروه آموزشی')
    },
    EducationGroupId: {
      displayName: displayName('گروه آموزشی'),
      numeric,
      required
    },
    EducationYearName: {
      displayName: displayName('سال تحصیلی')
    },
    EducationYearId: {
      displayName: displayName('سال تحصیلی'),
      numeric,
      required
    }
  }
};
